src/pages/index.js

```diff
@@ -90,44 +102,88 @@ const profileEditPopup = new PopupWithForm(
 profileEditPopup.setEventListeners();
 
 // Profile Edit Popup Button
+const profileEditButton = document.querySelector("#profile-edit-button");
+
 profileEditButton.addEventListener("click", () => {
-  const userInfoData = userInfo.getUserInfo();
+  if (userInfoInstance) {
+    const userInfoData = userInfoInstance.getUserInfo();
 
-  profileTitleInput.value = userInfoData.name;
-  profileDescriptionInput.value = userInfoData.job;
+    profileTitleInput.value = userInfoData.name;
+    profileDescriptionInput.value = userInfoData.job;
 
-  profileEditPopup.open();
+    profileFormValidator.resetValidation();
+    profileEditPopup.open();
+  }
 });
 
-// Profile Changes Save
 function handleProfileSaveSubmit(inputValues) {
-  userInfo.setUserInfo({
+  const saveButton = document.getElementById("profile-modal-save-button");
+
+  saveButton.textContent = "Saving...";
+
+  userInfoInstance.setUserInfo({
     name: inputValues.title,
     job: inputValues.description,
+    avatar: avatarImg.src,
   });
+
+  const newInfo = {
+    name: inputValues.title,
+    about: inputValues.description,
+    avatar: avatarImg.src,
+  };
+
+  apiUpdateProfileInfo(newInfo).then(() => (saveButton.textContent = "Save"));
 }
 
-// ************************** Preview Image ************************** //
+// *********************** Profile Avatar Edit *********************** //
 
-// Preview Image Popup
-const previewImagePopup = new PopupWithImage({
-  popupSelector: "#preview-image-modal",
-});
+const avatarModal = new PopupWithForm(
+  {
+    popupSelector: "#avatar-edit-modal",
+  },
+  handleAvatarFormSubmit
+);
 
-// Clicking an Image to get the preview Image
-function handleImageClick(cardData) {
-  previewImagePopup.open(cardData);
-}
+avatarModal.setEventListeners();
 
-previewImagePopup.setEventListeners();
+const avatarEditButton = document.getElementById("avatar-image");
 
-// Render Card Function
-const renderCard = (card) => {
-  const cardInstance = new Card(card, "#card-template", handleImageClick);
+avatarEditButton.addEventListener("click", () => {
+  editAvatarFormValidator.resetValidation();
 
-  const cardElement = cardInstance.generateCard();
-  cardSection.addItem(cardElement);
-};
+  avatarModal.open();
+});
+
+function handleAvatarFormSubmit(formData) {
+  const saveButton = document.getElementById("avatar-modal-save-button");
+
+  saveButton.textContent = "Saving...";
+
+  const avatarLink = formData.url;
+
+  const avatarImg = document.getElementById("avatar-image");
+  avatarImg.src = avatarLink;
+
+  api
+    .updateAvatar(avatarLink)
+    .then((updatedInfo) => {
+      userInfoInstance.setUserInfo({
+        name: updatedInfo.name,
+        job: updatedInfo.about,
+        avatar: updatedInfo.avatar,
+      });
+
+      avatarModal.close();
+    })
+    .catch((error) => {
+      console.error(error);
+    })
+    .finally(() => {
+      saveButton.textContent = "Save";
+      addCardPopup.close();
```

Comment:
`  addCardPopup.close();`

* You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well